### PR TITLE
Employee banking controller refactor & Unit tests

### PR DESCRIPTION
**Pull request by DewanAlberts**
image

### CODE REVIEW

File: RR.App.Tests/Controllers/HRIS/EmployeeBankingControllerUnitTests.cs [hunk 1]
```diff
@@ -151,65 +91,77 @@ public async Task GetValidStatusReturnsOkResultWithEntries()
...
 
     [Fact]
-    public async Task UpdateExceptionThrownReturnsNotFoundResultWithErrorMessage()
+    public async Task GetBankingUnauthorizedAccess()
     {
-        _employeeBankingServiceMock.Setup(x => x.Update(It.IsAny<EmployeeBankingDto>(), "test@example.com"))
-                   .ThrowsAsync(new Exception("Some error message"));
+        _identity.Setup(i => i.Role).Returns("Talent");
+        _identity.SetupGet(i => i.EmployeeId).Returns(5);
+        _identity.SetupGet(i => i.Email).Returns("talent@example.com");
```
Comment (Juanro-Minne) [inline RR.App.Tests/Controllers/HRIS/EmployeeBankingControllerUnitTests.cs]: maybe setup the _identity at class level
image

File: HRIS.Services/Services/EmployeeBankingService.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 using HRIS.Models.Enums;
 using HRIS.Services.Interfaces;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Update;
```
Comment (EduanO) [inline HRIS.Services/Services/EmployeeBankingService.cs]:
```suggestion
```

File: HRIS.Services/Services/EmployeeBankingService.cs [hunk 3]
```diff
@@ -37,6 +38,20 @@ public async Task<EmployeeBankingDto> Delete(int addressId)
...
+            BankName = newEntry.BankName,
+            Branch = newEntry.Branch,
+            AccountNo = newEntry.AccountNo,
+            AccountType = newEntry.AccountType,
+            Status = newEntry.Status,
+            DeclineReason = newEntry.DeclineReason,
+            File = newEntry.File,
+            LastUpdateDate = DateOnly.FromDateTime(DateTime.Now),
+            PendingUpdateDate = DateOnly.FromDateTime(DateTime.Now)
+        };
```
Comment (EduanO) [inline HRIS.Services/Services/EmployeeBankingService.cs]:
```suggestion
```

File: HRIS.Services/Services/EmployeeBankingService.cs [hunk 4]
```diff
@@ -108,6 +123,21 @@ public async Task<List<EmployeeBankingDto>> GetBanking(int employeeId)
...
+
```
Comment (EduanO) [inline HRIS.Services/Services/EmployeeBankingService.cs]:
```suggestion
```

File: RR.App.Tests/Controllers/HRIS/EmployeeBankingControllerUnitTests.cs [hunk 5]
```diff
@@ -2,10 +2,13 @@
 using HRIS.Models;
 using HRIS.Models.Enums;
 using HRIS.Services.Interfaces;
+using HRIS.Services.Services;
 using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.HttpResults;
```
Comment (EduanO) [inline RR.App.Tests/Controllers/HRIS/EmployeeBankingControllerUnitTests.cs]:
```suggestion
```

File: RR.App.Tests/Controllers/HRIS/EmployeeBankingControllerUnitTests.cs [hunk 6]
```diff
@@ -19,17 +22,15 @@ public class EmployeeBankingControllerUnitTests
     private readonly ClaimsPrincipal _claimsPrincipal;
     private readonly EmployeeBankingController _controller;
     private readonly ClaimsIdentity _claimsIdentity;
-
+    private readonly AuthorizeIdentityMock _identity;
```
Comment (EduanO) [inline RR.App.Tests/Controllers/HRIS/EmployeeBankingControllerUnitTests.cs]:
```suggestion
```

File: RR.App/Controllers/HRIS/EmployeeBankingController.cs [hunk 7]
```diff
@@ -22,128 +23,47 @@ public EmployeeBankingController(AuthorizeIdentity identity, IEmployeeBankingSer
...
-            return Ok(employeeBankingDto);
-        }
-        catch (Exception ex)
-        {
-            if (ex.Message.Contains("exists"))
-                return Problem("Unexceptable", "Unexceptable", 406, "Details already exists");
-            if (ex.Message.Contains("Unauthorized access")) return StatusCode(403, $"Forbidden: {ex.Message}");
-            return NotFound(ex.Message);
-        }
+        var claimsIdentity = User.Identity as ClaimsIdentity;
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/EmployeeBankingController.cs]:
```suggestion
```

File: RR.App/Controllers/HRIS/EmployeeBankingController.cs [hunk 8]
```diff
@@ -22,128 +23,47 @@ public EmployeeBankingController(AuthorizeIdentity identity, IEmployeeBankingSer
...
+        var employeeBankingDto =
+            await _employeeBankingService.Save(newEntry, claimsIdentity!.FindFirst(ClaimTypes.Email)!.Value);
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/EmployeeBankingController.cs]:
```suggestion
            await _employeeBankingService.Save(newEntry, _identity.Email);
```

File: RR.App/Controllers/HRIS/EmployeeBankingController.cs [hunk 9]
```diff
@@ -22,128 +23,47 @@ public EmployeeBankingController(AuthorizeIdentity identity, IEmployeeBankingSer
...
-            var claimsIdentity = User.Identity as ClaimsIdentity;
-            await _employeeBankingService.Update(Bankingdto, claimsIdentity!.FindFirst(ClaimTypes.Email)!.Value);
-
-            return Ok();
-        }
-        catch (Exception ex)
-        {
-            return NotFound(ex.Message);
-        }
+        var claimsIdentity = User.Identity as ClaimsIdentity;
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/EmployeeBankingController.cs]:
```suggestion
```

File: RR.App/Controllers/HRIS/EmployeeBankingController.cs [hunk 10]
```diff
@@ -22,128 +23,47 @@ public EmployeeBankingController(AuthorizeIdentity identity, IEmployeeBankingSer
...
+        await _employeeBankingService.Update(updateEntry, claimsIdentity!.FindFirst(ClaimTypes.Email)!.Value);
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/EmployeeBankingController.cs]:
```suggestion
        await _employeeBankingService.Update(updateEntry, _identity.Email);
```

File: RR.App/Controllers/HRIS/EmployeeBankingController.cs [hunk 11]
```diff
@@ -22,128 +23,47 @@ public EmployeeBankingController(AuthorizeIdentity identity, IEmployeeBankingSer
...
-                if (_identity.EmployeeId == id)
-                {
-                    var employeeBanking = await _employeeBankingService.GetBanking(id);
-                    return Ok(employeeBanking);
-                }
-            }
-
-            return NotFound("Tampering found!");
-        }
+        if (_identity.Role is not ("SuperAdmin" or "Admin") || _identity.EmployeeId != id)
```
Comment (EduanO) [inline RR.App/Controllers/HRIS/EmployeeBankingController.cs]:
```suggestion
        if (_identity.Role is not ("SuperAdmin" or "Admin") && _identity.EmployeeId != id)
```

**Review by EduanO — CHANGES_REQUESTED** [review]

**Review by EduanO — APPROVED** [review]

**Review by ReinoRabbit — APPROVED** [review]

**Review by Juanro-Minne — APPROVED** [review]

**Pull request merged**